h/api/views.py

```diff
@@ -156,15 +156,35 @@ def create(request):
     """Create an annotation from the POST payload."""
     json_payload = _json_payload(request)
 
+    # Validate the annotation for, and create the annotation in, PostgreSQL.
+    if request.feature('postgres'):
+        schema = schemas.CreateAnnotationSchema(request)
+        appstruct = schema.validate(copy.deepcopy(json_payload))
+        annotation = storage.create_annotation(request, appstruct)
+
+    # Validate the annotation for, and create the annotation in, Elasticsearch.
     legacy_schema = schemas.LegacyCreateAnnotationSchema(request)
     legacy_appstruct = legacy_schema.validate(copy.deepcopy(json_payload))
 
-    annotation = storage.create_annotation(request, legacy_appstruct)
+    # When 'postgres' is on make sure that annotations in the legacy
+    # Elasticsearch database use the same IDs as the PostgreSQL ones.
+    if request.feature('postgres'):
+        assert annotation.id
+        legacy_appstruct['id'] = annotation.id
 
-    _publish_annotation_event(request, annotation, 'create')
+    legacy_annotation = storage.legacy_create_annotation(request,
+                                                         legacy_appstruct)
```

Comment:
Just found a bug when I tried this out creating a reply to an annotation.

So the problem is that `legacy_create_annotation` ends up calling `h.api.transform.prepare` with a partial func for `h.api.storage.fetch_annotation`. Then inside the transform it fetches the top level annotation with that passed-in partial func and then expects the result to be a dictionay (`if 'group' in top_level_annotation:`), but since the postgres feature flag is on, the fetcher returns a postgres model instance.